python/pyarrow/jvm.py

```diff
@@ -28,24 +28,45 @@
 import pyarrow as pa
 
 
-def jvm_buffer(arrowbuf):
+class _JvmBufferNanny:
+    """
+    An object that keeps a org.apache.arrow.memory.ArrowBuf's underlying
+    memory alive.
+    """
+    ref_manager = None
```

Comment:
Hi @pitrou I was trying to understand why you used a static variable here instead of `jvm_buf.retain()` like before? I'm a little concerned about how this would behave with multiple threads and buffers that have different `ReferenceManager`s since `retain()` is called on the static var, but `release()` is called on the instance var..

Reply:
You are misunderstanding how this works. This is not C++, when you write:
```python
        self.ref_manager = ref_manager
```
it creates an instance attribute which shadows the class attribute.

Reply:
So if you have multiple threads creating `jvm_buffer`s with different backing reference manager instances, isn't it possible that one thread will overwrite the class attribute `ref_manager` being used in another thread, then both will call `ref_manager.retain()` using the same `ref_manager` class attribute?

Reply:
Hmm, no. The class attribute `ref_manager` will always remain `None`.

Reply:
Oh, ok. I think I was confused over having a local var `ref_manager` too. So why do you need a class attribute? Is it in case of an exception is thrown and the class instance doesn't get assigned?

Reply:
Ah, no, it's not needed. It's just a common idiom to write:
```python
class A:
    attr = None
```

instead of:
```python
class A:
    def __init__(self):
        self.attr = None
```

Reply:
So, yes, I could have written `self.ref_manager = None` inside the constructor instead. Sorry for the confusion.

Reply:
Ok, no problem, thanks for clearing that up for me.